src/Storages/StorageFile.cpp

```diff
@@ -875,466 +879,397 @@ static std::chrono::seconds getLockTimeout(ContextPtr context)
-                input_format = FormatFactory::instance().getInput(
-                    storage->format_name, *read_buf, block_for_format, context, max_block_size, storage->format_settings,
-                    max_parsing_threads, std::nullopt, /*is_remote_fs*/ false, CompressionMethod::None, need_only_count);
+            chassert(file_num > 0);
 
-                if (key_condition)
-                    input_format->setKeyCondition(key_condition);
+            const auto max_parsing_threads = std::max<size_t>(settings.max_threads / file_num, 1UL);
+            input_format = FormatFactory::instance().getInput(
+                storage->format_name, *read_buf, block_for_format, context, max_block_size, storage->format_settings,
+                max_parsing_threads, std::nullopt, /*is_remote_fs*/ false, CompressionMethod::None, need_only_count);
 
-                if (need_only_count)
-                    input_format->needOnlyCount();
+            if (key_condition)
+                input_format->setKeyCondition(key_condition);
 
-                QueryPipelineBuilder builder;
-                builder.init(Pipe(input_format));
+            if (need_only_count)
+                input_format->needOnlyCount();
 
-                if (columns_description.hasDefaults())
-                {
-                    builder.addSimpleTransform([&](const Block & header)
-                    {
-                        return std::make_shared<AddingDefaultsTransform>(header, columns_description, *input_format, context);
-                    });
-                }
+            QueryPipelineBuilder builder;
+            builder.init(Pipe(input_format));
 
-                /// Add ExtractColumnsTransform to extract requested columns/subcolumns
-                /// from chunk read by IInputFormat.
+            if (columns_description.hasDefaults())
+            {
                 builder.addSimpleTransform([&](const Block & header)
-                {
-                    return std::make_shared<ExtractColumnsTransform>(header, requested_columns);
-                });
+                                           {
+                                               return std::make_shared<AddingDefaultsTransform>(header, columns_description, *input_format, context);
+                                           });
+            }
 
-                pipeline = std::make_unique<QueryPipeline>(QueryPipelineBuilder::getPipeline(std::move(builder)));
-                reader = std::make_unique<PullingPipelineExecutor>(*pipeline);
+            /// Add ExtractColumnsTransform to extract requested columns/subcolumns
+            /// from chunk read by IInputFormat.
+            builder.addSimpleTransform([&](const Block & header)
+                                       {
+                                           return std::make_shared<ExtractColumnsTransform>(header, requested_columns);
+                                       });
```

Comment:
Indents. I understand that they were changed by the IDE while copying, just pay more attention to it